<a href="https://colab.research.google.com/github/noviansyah972/DataScience2026_240401010138_NoviansyahRamadhan/blob/main/Pertemuan2_NoviansyahRamadhan_240401010132.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Identitas Mahasiswa

**Nama Lengkap:** Noviansyah Ramadhan

**NIM:** 240401010132

**Kelas:** IF-403

In [2]:
import pandas as pd
import numpy as np

url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df  = pd.read_csv(url)

df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [3]:
# Q1 : Berapa jumlah baris dan kolom dataset Titanic
print (df.shape)

(891, 12)


Untuk mencari jumlah baris dan kolom pada dataset bisa menggunakan fungsi df.shape pada library pandas.

Pada output df.shape (891, 12) menunjukan bahwa jumlah baris dan kolom pada data csv adalah 891 baris dan 12 kolom

In [10]:
# Q2 Kolom mana yang memiliki nilai kosong (missing)?
print(df.isnull().sum())

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


Untuk mencari kolom yang memiliki nilai kosong (missing) bisa menggunakan fungsi df.isnull dan dikombinasikan dengan operasi sum untuk menjumlahkan data yang kosong pada setiap kolom.

Output yang di hasilkan dari kombinasi (df.isnull().sum()) akan menampilkan jumlah setiap data kosong (missing/null) pada kolom.



In [52]:
# Q3 Berapa persen penumpang yang selamat (Survived=1)?
survived_passenger = df['Survived'].sum()
survived_passenger_precentage  = df['Survived'].mean()*100

print(f"Penumpang yang selamat adalah {survived_passenger}")
print(f"Persentase penumpang yang selamat adalah {survived_passenger_precentage:.5f}%")


Penumpang yang selamat adalah 342
Persentase penumpang yang selamat adalah 38.38384%


Untuk mencari persentase penumpang yang selamat, bisa menjumlahkan dahulu kolom survived dengan fungsi

`df['Survived'].sum()`


lalu bisa melajutkan mencari rata rata (mean) antara data survived dengan seluruh data dan di kalikan 100 untuk mengubah ke persen

`df['Survived'].mean()*100`


setelah melakukan program tersebut, maka output bisa menggunakan variable yang sudah di buat

`print(f"Penumpang yang selamat adalah {survived_passenger}")`

`print(f"Persentase penumpang yang selamat adalah {survived_passenger_precentage:.5f}%")`


In [58]:
# Q4 Rata-rata usia penumpang wanita kelas 1 yang selamat?
wk1 = df[
  (df['Sex']=='female') & #Seleksi gender
  (df['Pclass']==1) &     #Seleksi kelas
  (df['Survived']==1)     #seleksi selamat/tidak
]
print (f"Rata-rata usia penumpang wanita kelas 1 yang selamat:{wk1['Age'].mean():.2f} tahun")

Rata-rata usia penumpang wanita kelas 1 yang selamat:34.94 tahun


Lakukan seleksi data terlebih dahulu menggunakan boolean indexing sebagai berikut

```
wk1 = df[
  (df['Sex']=='female') & #Seleksi gender
  (df['Pclass']==1) &     #Seleksi kelas
  (df['Survived']==1)     #seleksi selamat/tidak
]
```
Maka bisa di lanjutkan mencari rata-rata usia wanita kelas 1 yang selamat dan menjadikannya output

`print (f"Rata-rata usia penumpang wanita kelas 1 yang selamat:{wk1['Age'].mean():.2f} tahun")`


In [71]:
# Q5 Kelas penumpang mana yang tingkat keselamatannya tinggi?
hasil = df.groupby('Pclass').agg(
    total=('PassengerId', 'count'),         #jumlah penumpang berdasarkan kelas
    selamat=('Survived','sum'),             #jumlah penumpang selamat berdasarkan kelas
    tingkat_keselamatan=('Survived','mean') #perhitungan rata rata penumpang selamat dari jumlah penumpang
)
hasil['tingkat_keselamatan'] = hasil['tingkat_keselamatan'] * 100

print(hasil.round(2))


        total  selamat  tingkat_keselamatan
Pclass                                     
1         216      136                62.96
2         184       87                47.28
3         491      119                24.24


Mencari kelas yang memiliki tingkat keselamatan paling tinggi bisa menggunakan `df.groupby` untuk mengelompokkan data berdasarkan suatu kolom. pada case ini pengelompokan di lakukan pada kolom class,

dilanjutkan menggunakan aggregate untuk melakukan beberapa perhitungan sekaligus pada setiap kelompok.

In [73]:
# Q6 Gender penumpang yang memiliki tingkat keselamatan tinggi?
hasil = df.groupby('Sex').agg(
    total=('Sex', 'count'),         #jumlah penumpang berdasarkan gender
    selamat=('Survived','sum'),             #jumlah penumpang selamat kelas x
    tingkat_keselamatan=('Survived','mean') #perhitungan rata rata penumpang selamat dari jumlah penumpang berdasarkan gender
)
hasil['tingkat_keselamatan'] = hasil['tingkat_keselamatan'] * 100

print(hasil.round(2))

        total  selamat  tingkat_keselamatan
Sex                                        
female    314      233                74.20
male      577      109                18.89
